pkg/commands/workload_apply_test.go

```diff
@@ -8909,6 +8909,435 @@ To get status: "tanzu apps workload get my-workload"
+     10 + |  source:
+     11 + |    git:
+     12 + |      ref:
+     13 + |        branch: main
+     14 + |      url: my-repo-server/my-repo
+     15 + |    subPath: testdata/local-source/subpath
+👍 Updated workload "my-workload"
+
+To see logs:   "tanzu apps workload tail my-workload --timestamp --since 1h"
+To get status: "tanzu apps workload get my-workload"
+
+`,
+		},
+		{
+			Name: "update local source to git without changing subpath using flags",
+			Skip: runtm.GOOS == "windows",
+			Args: []string{workloadName, flags.GitBranchFlagName, "main", flags.GitRepoFlagName, "my-repo-server/my-repo", flags.YesFlagName},
+			GivenObjects: []client.Object{
+				parent.
+					MetadataDie(func(d *diemetav1.ObjectMetaDie) {
+						d.Annotations(map[string]string{apis.LocalSourceProxyAnnotationName: ":default-my-workload@sha256:978be33a7f0cbe89bf48fbb438846047a28e1298d6d10d0de2d64bdc102a9e69"})
+						d.Labels(map[string]string{apis.WorkloadTypeLabelName: "web"})
+					}).SpecDie(func(d *diecartov1alpha1.WorkloadSpecDie) {
+					d.Source(
+						&cartov1alpha1.Source{
+							Image:   ":default-my-workload@sha256:978be33a7f0cbe89bf48fbb438846047a28e1298d6d10d0de2d64bdc102a9e69",
+							Subpath: subpath,
+						},
+					)
+				}),
+			},
+			KubeConfigTransport: clitesting.NewFakeTransportFromResponse(respCreator(http.StatusOK, `{"statuscode": "200", "message": "any ignored message"}`, myWorkloadHeader)),
+			ExpectUpdates: []client.Object{
+				parent.
+					MetadataDie(func(d *diemetav1.ObjectMetaDie) {
+						d.Labels(map[string]string{apis.WorkloadTypeLabelName: "web"})
+					}).SpecDie(func(d *diecartov1alpha1.WorkloadSpecDie) {
+					d.Source(
+						&cartov1alpha1.Source{
+							Git: &cartov1alpha1.GitSource{
+								URL: "my-repo-server/my-repo",
+								Ref: cartov1alpha1.GitRef{
+									Branch: "main",
+								},
+							},
+							Subpath: subpath,
+						},
+					)
+				}),
+			},
+			ExpectOutput: `
+🔎 Update workload:
+  1,  1   |---
+  2,  2   |apiVersion: carto.run/v1alpha1
+  3,  3   |kind: Workload
+  4,  4   |metadata:
+  5     - |  annotations:
+  6     - |    local-source-proxy.apps.tanzu.vmware.com: :default-my-workload@sha256:978be33a7f0cbe89bf48fbb438846047a28e1298d6d10d0de2d64bdc102a9e69
+  7,  5   |  labels:
+  8,  6   |    apps.tanzu.vmware.com/workload-type: web
+  9,  7   |  name: my-workload
+ 10,  8   |  namespace: default
+ 11,  9   |spec:
+ 12, 10   |  source:
+ 13     - |    image: :default-my-workload@sha256:978be33a7f0cbe89bf48fbb438846047a28e1298d6d10d0de2d64bdc102a9e69
+     11 + |    git:
+     12 + |      ref:
+     13 + |        branch: main
+     14 + |      url: my-repo-server/my-repo
+ 14, 15   |    subPath: testdata/local-source/subpath
+👍 Updated workload "my-workload"
+
+To see logs:   "tanzu apps workload tail my-workload --timestamp --since 1h"
+To get status: "tanzu apps workload get my-workload"
+
+`,
+		},
+		{
+			Name: "update local source to image using flags",
```

Comment:
there is a similar test, the only difference is subpath. should we modify existing test or is it better to have a different one?

Reply:
Updated the existing one and removed the new one.